src/libmongoc/src/mongoc/mongoc-crypt.c

```diff
@@ -1063,52 +1007,20 @@ _mongoc_crypt_create_datakey (_mongoc_crypt_t *crypt,
       goto fail;
    }
 
-   if (0 == strcmp ("aws", kms_provider) && masterkey) {
```

Comment:
Similarly, `mongocrypt_ctx_setopt_key_encryption_key` consolidates the configuration for the master key. So we can pass a BSON document of the options with the "provider" field appended.